openssl/src/sign.rs

```diff
@@ -139,25 +139,48 @@ impl<'a> Signer<'a> {
         }
     }
 
-    pub fn finish(&self) -> Result<Vec<u8>, ErrorStack> {
+    /// Computes an upper bound on the signature length.
+    pub fn finish_len(&self) -> Result<usize, ErrorStack> {
         unsafe {
             let mut len = 0;
             cvt(ffi::EVP_DigestSignFinal(
                 self.md_ctx,
                 ptr::null_mut(),
                 &mut len,
             ))?;
-            let mut buf = vec![0; len];
+            Ok(len)
+        }
+    }
+
+    /// Outputs the signature into the provided buffer, returning the
+    /// length of that buffer.
+    ///
+    /// This method will fail if the buffer is not large enough for
+    /// the signature, one can use `finish_len` to get an upper bound
+    /// on the required size.
+    pub fn finish_into<B:AsMut<[u8]>>(&self, mut buf: B) -> Result<usize, ErrorStack> {
```

Comment:
Let's just have `buf` be a `&mut [u8]`.

Reply:
I'm curious, why?

Reply:
It simplifies the function signature and speeds up compilation a bit by avoiding a bunch of copies of the method. Similar methods in the standard library take `&mut [u8]` as well. You're either already going to have a `&mut [u8]` or a `Vec<u8>`, which coerces into a slice.

Reply:
Ok, done! Thanks for taking the time to explain.